Automattic/wp-calypso
# Implement GA4 analytics for Jetpack Green

- Add gtag ts types
- Remove redundant initializers for gtag
- Assign a valid return type to cost-to-usd
- Do not change properties in the cart project
- Flag the function with the comment about wrong type of fn property
- Implement GA4
- Implement typed version of existing function

#### Proposed Changes

* Implement GA4 methods and enable it for Calypso Green
* Improve typing in analytics library in wp-calypso

#### Testing Instructions

* Install and enable [GA Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) in your Chrome

* Run Calypso Green with `yarn start-jetpack-cloud-p`
* Enable ad tracking on your local instance (PCYsg-dGP-p2#calypso)
* Go to `http://jetpack.cloud.localhost:3001/` and navigate through pages
* Go to the "Jetpack.com - GA4" property in the Google Analytics dashboard and navigate to "Configure > Debug view"
* You should see fired `page_view` events on the timeline

(optional)

* Run Calypso Blue with `yarn start` - remember to enable ad tracking (PCYsg-dGP-p2#calypso)
* Enter the checkout process; e.g. http://calypso.localhost:3000/checkout/jetpack/jetpack_personal/
* Navigate through the process, and pay for the product by the credit card (no WP credits) **
* You should see `purchase` event on the timeline in Debug View in GA

** Alternatively, you can change `cart.total_cost >= 0.01` condition in `record-purchase.js:7` to `true`
